pandas-stubs/_testing/__init__.pyi

```diff
@@ -1,65 +1,117 @@
-from collections.abc import Generator
+from collections.abc import (
+    Container,
+    Generator,
+    Iterable,
+)
 from contextlib import contextmanager
 from typing import (
     Any,
     Literal,
     overload,
 )
+import warnings
 
+from matplotlib.artist import Artist
+import numpy as np
 from pandas import (
+    Categorical,
     DataFrame,
     Index,
     Series,
 )
+from pandas.arrays import (
+    DatetimeArray,
+    IntervalArray,
+    PeriodArray,
+    SparseArray,
+    TimedeltaArray,
+)
+from pandas.core.arrays.base import ExtensionArray
+
+from pandas._typing import (
+    AnyArrayLike,
+    T,
+)
 
 def assert_almost_equal(
-    left,
-    right,
-    check_dtype: bool | str = ...,
-    check_less_precise: bool | int = ...,
+    left: T,
+    right: T,
```

Comment:
These two parameters could be any object but they have to have the same type. This is represented in the type system by using the same type variable for both parameters .